loader/loader.go

```diff
@@ -170,6 +173,42 @@ func (l *loader) GetMatchingReleasePlans(ctx context.Context, cli client.Client,
 	return releasePlans, nil
 }
 
+// GetPreviousRelease returns the Release that was created just before the given Release.
+// If no previous Release is found, a NotFound error is returned.
+func (l *loader) GetPreviousRelease(ctx context.Context, cli client.Client, release *v1alpha1.Release) (*v1alpha1.Release, error) {
+	releases := &v1alpha1.ReleaseList{}
+	err := cli.List(ctx, releases,
+		client.InNamespace(release.Namespace),
+		client.MatchingFields{"spec.releasePlan": release.Spec.ReleasePlan})
+	if err != nil {
+		return nil, err
+	}
+
+	var previousRelease *v1alpha1.Release
+
+	// Find the previous release
+	for i, possiblePreviousRelease := range releases.Items {
+		// Ignore the release passed as argument and any release created after that one
+		if possiblePreviousRelease.Name == release.Name ||
+			possiblePreviousRelease.CreationTimestamp.After(release.CreationTimestamp.Time) {
```

Comment:
Just a quick question here. I remember happy having trouble with this functionality as eventually you would have more than one release created at the very same second. So IIUC the `.After`  can (somehow) identify which one is the newer one.

Reply:
For the same app? Hard to see. In any case nothing we can do there. Any of it would be the winner and that's fine.